src/Riverline/MultiPartParser/StreamedPart.php

```diff
@@ -141,8 +141,7 @@ public function __construct($stream)
                         $currentOffset = ftell($this->stream);
                         // Get end of line length (should be 2)
                         $eofLength = strlen($line) - strlen($trimmed);
-                        // Part length is current offset - part start offset - separator length - (2 x eof length)
-                        $partLength = $currentOffset - $partOffset - strlen($separator) - (2 * $eofLength);
```

Comment:
This is the mail issue. 
Previously the code was using `strlen($separator)`, now it uses `strlen($trimmed)`. 
This is because the last part has two more `--` and was always making the last part two bytes longer.
Now the length of the last part is correct, This test https://github.com/Riverline/multipart-parser/pull/21/files#diff-095af6be0817b4f602a10c30c376ffa5R77 check is